dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/hibernate/HibernatePotentialDuplicateStore.java

```diff
@@ -165,24 +167,45 @@ public boolean exists( PotentialDuplicate potentialDuplicate )
     public void moveTrackedEntityAttributeValues( TrackedEntityInstance original, TrackedEntityInstance duplicate,
         List<String> trackedEntityAttributes )
     {
+        // Collect existing teav from original for the tea list
+        Map<String, TrackedEntityAttributeValue> originalAttributeValueMap = new HashMap<>();
+        original.getTrackedEntityAttributeValues().forEach( oav -> {
+            if ( trackedEntityAttributes.contains( oav.getAttribute().getUid() ) )
+            {
```

Comment:
Not sure about the standards here, but can we go like 

```
        Map<String, TrackedEntityAttributeValue> originalAttributeValueMap = new HashMap<String, TrackedEntityAttributeValue>() {
            {
                original.getTrackedEntityAttributeValues().forEach(oav -> {
                    if (trackedEntityAttributes.contains(oav.getAttribute().getUid())) {
                        put(oav.getAttribute().getUid(), oav);
                    }
                });
            }
        };
```
In any case i'd leave HashMap<String, TrackedEntityAttributeValue> for java 8 compatibility

Reply:
@lucaCambi77 Wdym Java 8 compatibility? This type of assignment type inference is supported even in Java 7. So surely in Java 8 as well.
Map<String, TrackedEntityAttributeValue> originalAttributeValueMap = new HashMap<>();

Also regarding the other anonymous class way of initializing hashmaps, I dont think we have followed that way in most cases, so sticking to the plain old technique.

Reply:
@lucaCambi77 Wdym Java 8 compatibility? This type of assignment type inference is supported even in Java 7. So surely in Java 8 as well.
Map<String, TrackedEntityAttributeValue> originalAttributeValueMap = new HashMap<>();

Also regarding the other anonymous class way of initializing hashmaps, I dont think we have followed that way in most cases, so sticking to the plain old technique. That technique also creates an extra anonymous class at every usage, so the traditional approach is alright imo.